python/numpy.h

```diff
@@ -49,16 +49,15 @@ template <class T>
 auto cast_to_array_like(const py::object &obj, const units::Unit unit) {
   using TM = ElementTypeMap<T>;
   using PyType = typename TM::PyType;
+  TM::check_assignable(obj, unit);
```

Comment:
@jl-wynen I believe this change was from your branch? Can you confirm that should indeed also be applied to the final `else` branch?

Reply:
Yes, this should always be checked. But it does not do anything except for `time_point`. So there is no problem in practice with the original but it is cleaner to always check.